### ISSUE TITLE
max-nesting-depth not being picked up?

### ISSUE THREAD

**Issue by benfrain** [davidtheclark/stylelint-statement-max-nesting-depth#9]
Just trying to integrate this but doesn't seem to be working. Really hope I'm not wasting your time!

I created a reduction. Here is my package.json:

```
{
  "name": "stylelinttest",
  "version": "1.0.0",
  "description": "Testing stylelint",
  "main": "styles.css",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "author": "ben frain",
  "license": "ISC",
  "devDependencies": {
    "stylelint": "^4.2.0",
    "stylelint-statement-max-nesting-depth": "^2.0.1"
  }
}
```

Here is my test `.stylelintrc`:

```
{
  "plugins": [
    "stylelint-statement-max-nesting-depth"
  ],
  "rules": {
    "block-no-empty": true,
    "statement-max-nesting-depth": [2, { countAtRules: false }]
  }
}
```

here is styles.css:

```
.g5-ArrowButton {
    .g5-DrrowDutton .g5-SrrowDutton & {
        width: 100%;
    }
}

.thing .thing .thing {
    width: 100%;
}

.thing {

}
```

When I run `stylelint styles.css` I get this output:

```
styles.css
11:8    Unexpected empty block (block-no-empty)
```

I expected to see an error regarding the nesting depth. I'm obviously doing something incorrect?

**Comment by davidtheclark** [comment]
@benfrain. I see two things: 

If your config is JSON, shouldn't `countAtRules` have quotation marks around it? But that is unimportant, because when I added quotes I still did reproduce what you're seeing.

I think the misunderstanding here is about what the primary option value means. According to the plugin's expectations, your example CSS file _does not exceed a nesting depth of 1_ -- so it is not in fact violating your config. If you change your max to 0, you should see an error. Or if you nest to a depth of 3.

I tried to document the nesting levels here: https://github.com/davidtheclark/stylelint-statement-max-nesting-depth#details. Do those docs make sense now, or do you think there's another way I should try to explain?

**Comment by davidtheclark** [comment]
To add on: nesting to a depth of 3, which would violate your config, looks like this:

``` css
.foo {
  .bar { /* 1 */
    .baz { /* 2 */
      a { /* 3 */
```

**Comment by benfrain** [comment]
Ah - yes, my bad. I think it's my expectations and not your documentation! I was thinking it would effectively check for selector specificity so for example it would lint any rules that hit a specificity of 3 for example:

| inline | ID | Class | type |
| --- | --- | --- | --- |
| 0 | 0 | 3 | 0 |

I think I need a different plugin! ;)

**Comment by davidtheclark** [comment]
Yeah, that's a good plugin idea. Might make a good core rule for stylelint, in fact.

**Comment by benfrain** [comment]
Yes, that would be amazing if in core. That's generally what I need to police and kind of deals with the nesting issue too – however they write an overly specific selector we get them! :)

**Comment by davidtheclark** [comment]
Please file an issue over in stylelint. I think it would be a good addition, and it applies to standard CSS, unlike nesting depth, so qualifies as a core rule.

**Comment by benfrain** [comment]
Done: https://github.com/stylelint/stylelint/issues/721
